### PR TITLE
On Darwin, allow XCTest to be missing if we're only building swift-testing tests.

### PR DESCRIPTION
**Pull request by grynspan**
This PR removes the constraint on Darwin that XCTest.framework must be present in order to build tests using swift-testing. On Darwin, XCTest is included as a framework inside Xcode, but if a developer installs the Xcode Command Line Tools instead of the full IDE, XCTest is not included. They then get a diagnostic of the form:

> error: XCTest not available: terminated(1): /usr/bin/xcrun --sdk macosx --show-sdk-platform-path output:
>    xcrun: error: unable to lookup item 'PlatformPath' from command line tools installation
>    xcrun: error: unable to lookup item 'PlatformPath' in SDK '/Library/Developer/CommandLineTools/SDKs/MacOSX14.0.sdk'

Which is a poor experience if they aren't even using XCTest.

This change, as a (positive) side effect, suppresses the same diagnostic when running commands that are not usually dependent on the presence of XCTest such as `swift build`.

Note that swift-corelibs-xctest is not supported on Darwin, so installing the Xcode Command Line Tools and adding an explicit dependency on swift-corelibs-xctest will not produce a functional test target bundle. Supporting swift-corelibs-xctest on Darwin is a potential future direction.

Automated testing for this change is difficult because it relies on a build environment that is not supported in CI (namely the presence of the CL tools but not Xcode nor XCTest.framework.) I have manually tested the change against swift-testing's own test target.

A separate PR will be necessary in swift-testing to remove some remaining XCTest dependencies. Those changes are not covered by this PR.

Resolves rdar://125372431.

### CODE REVIEW

**Comment by grynspan** [comment]
@swift-ci please test

**Review by briancroom — APPROVED** [review]

File: Sources/Commands/SwiftTestCommand.swift [hunk 1]
```diff
@@ -814,7 +819,7 @@ final class TestRunner {
         #if os(macOS)
         if library == .xctest {
             guard let xctestPath = self.toolchain.xctestPath else {
-                throw TestError.xctestNotAvailable
+                throw TestError.xctestNotAvailable()
```
Comment (MaxDesiatov) [inline Sources/Commands/SwiftTestCommand.swift:822]: Why not provide a reason here for users to have an actionable error message?

File: Sources/PackageModel/SwiftSDKs/SwiftSDK.swift [hunk 2]
```diff
@@ -147,7 +147,29 @@ public struct SwiftSDK: Equatable {
...
+    @available(*, deprecated, message: "Use `xctestSupport` instead")
+    public var supportsTesting: Bool {
+        if case .supported = xctestSupport {
+            return true
+        }
+        return false
+    }
+
+    /// Whether or not the receiver supports testing using XCTest.
+    public enum XCTestSupport: Sendable, Equatable {
```
Comment (MaxDesiatov) [inline Sources/PackageModel/SwiftSDKs/SwiftSDK.swift:159]:
```suggestion
    package enum XCTestSupport: Sendable, Equatable {
```

File: Sources/PackageModel/SwiftSDKs/SwiftSDK.swift [hunk 3]
```diff
@@ -417,19 +439,44 @@ public struct SwiftSDK: Equatable {
...
+            hostTriple: hostTriple,
+            targetTriple: targetTriple,
+            toolset: toolset,
+            pathsConfiguration: pathsConfiguration,
+            xctestSupport: xctestSupport
+        )
+    }
+
     /// Creates a Swift SDK with the specified properties.
     public init(
```
Comment (MaxDesiatov) [inline Sources/PackageModel/SwiftSDKs/SwiftSDK.swift:468]:
```suggestion
    package init(
```

Comment (MaxDesiatov) [reply Sources/PackageModel/SwiftSDKs/SwiftSDK.swift:468]: Let's keep the new initializer hidden for now. We need to understand the exact use cases for exposing `SwiftSDK` as `public` in the first place before encouraging further adoption of it, especially as its API is not stable.

File: Sources/Commands/SwiftTestCommand.swift (hunk 1 above)
Comment (grynspan) [reply Sources/Commands/SwiftTestCommand.swift]: Done.

**Comment by grynspan** [comment]
@swift-ci please test

**Comment by grynspan** [comment]
@swift-ci please test Windows

**Review by MaxDesiatov — APPROVED** [review]
Thanks!

**Pull request merged**